v2/pkg/additional/local_stored_collector.go

```diff
@@ -116,14 +112,23 @@ func isImageByDigest(imgRef string) bool {
 	return strings.Contains(imgRef, "@")
 }
 
-func imageName(imgRef string) string {
-	var imageName string
-	imgSplit := strings.Split(imgRef, "@")
-	if len(imgSplit) > 1 {
-		imageName = imgSplit[0]
+func imagePathComponent(imgRef string) (string, error) {
+
+	var imageName []string
+	if isImageByDigest(imgRef) {
+		imageNameSplit := strings.Split(imgRef, "@")
+		imageName = strings.Split(imageNameSplit[0], "/")
+	} else {
+		imageName = strings.Split(imgRef, "/")
 	}
 
-	return imageName
+	if len(imageName) > 2 {
```

Comment:
@aguidirh - nice guarding against panics :)

Reply:
The tests are not affected by this change.